extensions/librdkafka/PublishKafka.cpp

```diff
@@ -79,6 +79,7 @@ core::Property PublishKafka::KerberosKeytabPath("Kerberos Keytab Path",
 core::Property PublishKafka::MessageKeyField("Message Key Field", "The name of a field in the Input Records that should be used as the Key for the Kafka message.\n"
                                              "Supports Expression Language: true (will be evaluated using flow file attributes)",
                                              "");
+core::Property PublishKafka::DebugContexts("Debug contexts", "A comma-separated list of debug contexts to enable. Including: generic, broker, topic, metadata, feature, queue, msg, protocol, cgrp, security, fetch, interceptor, plugin, consumer, admin, eos, all", "");
```

Comment:
Is this a defined list that can be controlled via allowable values?

Reply:
yes, according to [https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md](this)

Reply:
So then can you use our allowable values construct?

Reply:
I'm afraid not, the possible value can be any combination from the list above, not just a single element.

Reply:
Allowable values don't preclude having more than one option. Is there something that gave you that impression in the code?

Reply:
I assume we are talking about 'withAllowableValues' in PropertyBuilder. In 'LogAttribute.cpp', there is a set of allowable values ("debug", "info", etc), though only one of them can be set at a time. In this case, "msg,topic,broker",  "topic,msg" or "broker,msg" are all valid values. Thus, I don't understand how can allowable values can check that. If I misunderstand anything, I would love to hear your correction.

Reply:
@nghiaxlee what you say makes perfect sense, NiFI provides ListValidator for the given purpose:
https://github.com/apache/nifi/blob/master/nifi-commons/nifi-utils/src/main/java/org/apache/nifi/processor/util/StandardValidators.java#569

Although we don't have this functionality in MiNiFi at the moment, will create a follow-up Jira soon. I think the scope of this ticket is unquestionably small to cover this change.

Reply:
In this case you would not require a comma separated list and would simply create that comma separated list further down.

Reply:
I don't think allowable values limits you to a single resulting value. Can you not build a flowfile with the YAML multiple entries?

Reply:
@nghiaxlee I think I'm the long pole here so since my time is so fragmented. I created https://issues.apache.org/jira/browse/MINIFICPP-988 as a follow on with no release since there seems to be a disagreement about the approach. You are free to make a decision in that ticket, but I ask that you evaluate all options holistically.